### PR TITLE
upgrade rust dependencies

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: cli/op_error.rs [hunk 1]
```diff
@@ -261,11 +261,11 @@ impl From<&ReadlineError> for OpError {
   fn from(error: &ReadlineError) -> Self {
     use ReadlineError::*;
     let kind = match error {
-      Io(err) => return err.into(),
+      Io(err) => return OpError::from(err),
       Eof => ErrorKind::UnexpectedEof,
       Interrupted => ErrorKind::Interrupted,
       #[cfg(unix)]
-      Errno(err) => return err.into(),
+      Errno(_err) => todo!("why doesn't this work: (*err).into()"),
```
Comment (ry) [inline cli/op_error.rs]: I wasn't able to get this to work... not sure why

File: Cargo.lock [hunk 2]
```diff
@@ -988,6 +974,17 @@ version = "1.0.1"
...
+[[package]]
+name = "idna"
+version = "0.1.5"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "38f09e0f0b1fb55fdee1f17470ad800da77af5186a1a76c026b679358b7e844e"
+dependencies = [
+ "matches",
+ "unicode-bidi",
+ "unicode-normalization",
+]
```
Comment (ry) [inline Cargo.lock]: This is caused by sourcemap depending on an old version of the URL crate. I've send a PR upstream: https://github.com/getsentry/rust-sourcemap/pull/26

File: cli/op_error.rs [hunk 3]
```diff
@@ -261,11 +261,11 @@ impl From<&ReadlineError> for OpError {
   fn from(error: &ReadlineError) -> Self {
     use ReadlineError::*;
     let kind = match error {
-      Io(err) => return err.into(),
+      Io(err) => return OpError::from(err),
```
Comment (piscisaureus) [inline cli/op_error.rs]: What changed? I don't understand how this is different from before.

File: cli/op_error.rs [hunk 4]
```diff
@@ -261,11 +261,11 @@ impl From<&ReadlineError> for OpError {
...
       Eof => ErrorKind::UnexpectedEof,
       Interrupted => ErrorKind::Interrupted,
       #[cfg(unix)]
-      Errno(err) => return err.into(),
+      Errno(err) => return (*err).into(),
```
Comment (piscisaureus) [inline cli/op_error.rs]: And here?

File: cli/op_error.rs (hunk 3 above)
Comment (ry) [reply cli/op_error.rs]: Nothing just sloppy editing

**Pull request merged**